src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py

```diff
@@ -243,9 +243,9 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
     register_cli_argument('network nic {}'.format(item), 'load_balancer_inbound_nat_rule_ids', options_list=('--lb-inbound-nat-rules',), nargs='+', validator=validate_inbound_nat_rule_id_list, help='Space separated list of names or IDs of load balancer inbound NAT rules to associate with the NIC. If names are used, --lb-name must be specified.', completer=get_lb_subresource_completion_list('inboundNatRules'))
 
 # NIC ScaleSet
-register_cli_argument('network nic scale-set', 'virtual_machine_scale_set_name', options_list=('--vm-scale-set',), completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachineScaleSets'), id_part='name')
-register_cli_argument('network nic scale-set', 'virtualmachine_index', options_list=('--vm-index',))
-register_cli_argument('network nic scale-set', 'network_interface_name', id_part='child_name')
+register_cli_argument('vmss nic', 'virtual_machine_scale_set_name', options_list=('--vmss-name',), help='Scale set name.', completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachineScaleSets'), id_part='name')
+register_cli_argument('vmss nic', 'virtualmachine_index', options_list=('--vm-index',))
```

Comment:
you can choose to do it at other time, but to be consistent with the existing vmss command, ,this should be renamed to `--instance-id`, which is what compute team recommended